core/LSU.cpp

```diff
@@ -109,22 +98,48 @@ namespace olympia
     }
 
     // Receive new load/store instruction from Dispatch Unit
-    void LSU::getInstsFromDispatch_(const InstPtr & inst_ptr)
+    void LSU::getInstsFromDispatch_(const InstPtr &inst_ptr)
     {
-        core_types::RegFile reg_file = core_types::RF_INTEGER;
-        const auto & srcs = inst_ptr->getSourceOpInfoList();
-        if(srcs.size() > 0){
-            reg_file = olympia::coreutils::determineRegisterFile(srcs[0]);
+        bool all_ready = true; // assume all ready
+        if (!scoreboard_views_[core_types::RF_INTEGER]->isSet(inst_ptr->getSrcRegisterBitMask(core_types::RF_INTEGER)))
```

Comment:
Pls move your comment that you are checking address operands above the `if` statement

Reply:
Comments added in [5212a4c](https://github.com/riscv-software-src/riscv-perf-model/pull/70/commits/5212a4c3274e758abca7bebf8ebc45a4ac210ca9)